ksql-rest-app/src/main/java/io/confluent/ksql/rest/server/KsqlRestConfig.java

```diff
@@ -15,24 +15,50 @@
 
 package io.confluent.ksql.rest.server;
 
+import com.google.common.annotations.VisibleForTesting;
+import io.confluent.ksql.configdef.ConfigValidators;
 import io.confluent.ksql.rest.DefaultErrorMessages;
 import io.confluent.ksql.rest.ErrorMessages;
 import io.confluent.ksql.util.KsqlException;
+import io.confluent.ksql.util.KsqlServerException;
 import io.confluent.rest.RestConfig;
+import java.net.InetAddress;
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.net.UnknownHostException;
+import java.util.List;
 import java.util.Map;
+import java.util.Optional;
+import java.util.function.Function;
 import org.apache.kafka.common.config.ConfigDef;
 import org.apache.kafka.common.config.ConfigDef.Importance;
 import org.apache.kafka.common.config.ConfigDef.Type;
+import org.apache.kafka.common.config.ConfigException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 
 public class KsqlRestConfig extends RestConfig {
 
+  private static final Logger LOGGER = LoggerFactory.getLogger(KsqlRestConfig.class);
+
   private static final String KSQL_CONFIG_PREFIX = "ksql.";
 
   private static final String COMMAND_CONSUMER_PREFIX  =
       KSQL_CONFIG_PREFIX + "server.command.consumer.";
   private static final String COMMAND_PRODUCER_PREFIX  =
       KSQL_CONFIG_PREFIX + "server.command.producer.";
 
+  public static final String INTER_NODE_LISTENER_CONFIG =
+      KSQL_CONFIG_PREFIX + "inter.node.listener";
+  private static final String INTER_NODE_LISTENER_DOC =
+      "The listener used for communication between KSQL nodes in the cluster, if different to the '"
+          + LISTENERS_CONFIG + "' config property. "
+          + "In IaaS environments, this may need to be different from the interface to which "
+          + "the server binds. If this is not set, the first value from listeners will be used. "
+          + "Unlike listeners, it is not valid to use the 0.0.0.0 (IPv4) or [::] (IPv6) "
+          + "wildcard addresses.";
+  private static final String USE_FIRST_LISTENER = "http://use.first.listener";
```

Comment:
Do you need to have a default value that you don't intend to use as a default?  Seems like you could check for it being null or unset if that's the intention.

Reply:
Let me have another play. I actually intended to fix this before submitting, but forgot.